### PR TITLE
CMP-2458: PCI-DSS 4 Requirement 6

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Requirements `6.1`, `6.2` and `6.5` are `not applicable`
- Requirements `6.3` and `6.4` are inheremently and partially met.

### CODE REVIEW

**Comment by yuumasato** [comment]
@mildas @ggbecker @Mab879 Have you guys noticed increased flakiness with `Gate / Build, Test on CentOS 7 (Container) (pull_request)` test?
I feel like my PRs have been failing on unrelated problems...

File: applications/openshift/confinement/security_profiles_operator_exists/rule.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+title: "Make sure the Security Profiles Operator is installed"
+
+description: |-
+    Security Profiles Operator provides a way to define secure computing (seccomp) profiles and
+    SELinux profiles as custom resources that are syncrhonized to every node in a given namespace.
+
+    Using security profiels can increase security at the container level in your cluster.
```
Comment (rhmdnd) [inline applications/openshift/confinement/security_profiles_operator_exists/rule.yml:7]: nit: profiles*

File: applications/openshift/confinement/security_profiles_operator_exists/rule.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Seccomp security profiles list the syscalls a process can make, and SELinux security profiles
+    provide a label-based system taht restricts access and usage of processes, applications, and
```
Comment (rhmdnd) [inline applications/openshift/confinement/security_profiles_operator_exists/rule.yml:9]: nit: that

File: controls/pcidss_4_ocp4.yml [hunk 3]
```diff
@@ -1474,35 +1474,40 @@ controls:
...
       levels:
         - base
-      status: pending
+      status: not applicable
       notes: |-
-        Examine documentation and interview personnel to verify that security policies and
-        operational procedures identified in Requirement 6 are managed in accordance with all
-        elements specified in this requirement.
+        The responsibility for documentation, maintenance, use and dissemination of the processes
+        and mechanismis for developing and maintaining secure systems and software is on the
```
Comment (rhmdnd) [inline controls/pcidss_4_ocp4.yml:1487]: nit: mechanisms*

File: controls/pcidss_4_ocp4.yml [hunk 4]
```diff
@@ -1474,35 +1474,40 @@ controls:
...
     title: Bespoke and custom software are developed securely.
     levels:
       - base
-    status: pending
+    status: not applicable
+    notes: |-
+      OpenShift is developed and maintained following secure software development practices:
+      https://www.redhat.com/en/topics/security/red-hat-sdl
+      But this requirement applies only to software developed for or by the entity for the
+      entity's own use. This does not apply to third-party sofware.
```
Comment (rhmdnd) [inline controls/pcidss_4_ocp4.yml:1510]: nit: software

File: controls/pcidss_4_ocp4.yml [hunk 5]
```diff
@@ -1554,27 +1559,41 @@ controls:
...
+        - Risk rankings identify, at a minimum, all vulnerabilities considered to be a high-risk
+        or critical to the environment.
+        - Vulnerabilities for bespoke and custom, and third-party software (for example operating
+        systems and databases) are covered.
       levels:
         - base
-      status: pending
+      status: not applicable
+      notes: |-
+        The payment entity needs to stablish its own process of monitoring for vulnerabilities for
```
Comment (rhmdnd) [inline controls/pcidss_4_ocp4.yml:1595]: nit: establish*

**Review by rhmdnd — COMMENTED** [review]
Looks good, only minor spelling corrections.

**Comment by xiaojiey** [comment]
@yuumasato For rule, there is no compliance.openshift.io/automated-remediation= label, and the auto remediation cannot  get applied. Could you please help to check? Thanks.
```
##1. for a cluster where security profiles operator and container security operator not installed. Seems the auto-remediation for security profiles operators cannot be applied:
% cat ssb_pci.yaml        
apiVersion: compliance.openshift.io/v1alpha1
kind: ScanSettingBinding
metadata:
  name: pci-dss-compliance
  namespace: openshift-compliance
profiles:
  - name: upstream-ocp4-pci-dss-4-0
    kind: Profile
    apiGroup: compliance.openshift.io/v1alpha1
  - name: upstream-ocp4-pci-dss-node-4-0
    kind: Profile
    apiGroup: compliance.openshift.io/v1alpha1
settingsRef:
  name: default-auto-apply
  kind: ScanSetting
  apiGroup: compliance.openshift.io/v1alpha1
% oc apply -f ssb_pci.yaml
scansettingbinding.compliance.openshift.io/pci-dss-compliance created
% oc get ccr | grep FAIL
upstream-ocp4-pci-dss-4-0-acs-sensor-exists                                                    FAIL     medium
upstream-ocp4-pci-dss-4-0-api-server-encryption-provider-cipher                                FAIL     medium
upstream-ocp4-pci-dss-4-0-audit-log-forwarding-enabled                                         FAIL     medium
upstream-ocp4-pci-dss-4-0-audit-profile-set                                                    FAIL     medium
upstream-ocp4-pci-dss-4-0-container-security-operator-exists                                   FAIL     medium
upstream-ocp4-pci-dss-4-0-kubeadmin-removed                                                    FAIL     medium
upstream-ocp4-pci-dss-4-0-ocp-allowed-registries                                               FAIL     medium
upstream-ocp4-pci-dss-4-0-ocp-allowed-registries-for-import                                    FAIL     medium
upstream-ocp4-pci-dss-4-0-security-profiles-operator-exists                                    FAIL     medium
% oc get cr
NAME                                                              STATE
upstream-ocp4-pci-dss-4-0-api-server-encryption-provider-cipher   Applied
upstream-ocp4-pci-dss-4-0-audit-profile-set                       Applied
% oc get ccr upstream-ocp4-pci-dss-4-0-security-profiles-operator-exists --show-labels
NAME                                                          STATUS   SEVERITY   LABELS
upstream-ocp4-pci-dss-4-0-security-profiles-operator-exists   FAIL     medium     compliance.openshift.io/check-severity=medium,compliance.openshift.io/check-status=FAIL,compliance.openshift.io/profile-guid=382ba151-4843-575f-878e-4f77b81357b0,compliance.openshift.io/scan-name=upstream-ocp4-pci-dss-4-0,compliance.openshift.io/suite=pci-dss-compliance
#####After security profiles operator and container security operator installed:
% oc annotate  compliancescan upstream-ocp4-pci-dss-4-0 compliance.openshift.io/rescan=
compliancescan.compliance.openshift.io/upstream-ocp4-pci-dss-4-0 annotated
% oc get scan -w                                                                       
NAME                                    PHASE     RESULT
upstream-ocp4-pci-dss-4-0               RUNNING   NOT-AVAILABLE
upstream-ocp4-pci-dss-node-4-0-master   DONE      COMPLIANT
upstream-ocp4-pci-dss-node-4-0-worker   DONE      COMPLIANT
upstream-ocp4-pci-dss-4-0               AGGREGATING   NOT-AVAILABLE
upstream-ocp4-pci-dss-4-0               AGGREGATING   NOT-AVAILABLE
upstream-ocp4-pci-dss-4-0               DONE          NON-COMPLIANT
^C% % oc get ccr | grep FAIL
upstream-ocp4-pci-dss-4-0-acs-sensor-exists                                                    FAIL     medium
upstream-ocp4-pci-dss-4-0-audit-log-forwarding-enabled                                         FAIL     medium
upstream-ocp4-pci-dss-4-0-kubeadmin-removed                                                    FAIL     medium
upstream-ocp4-pci-dss-4-0-ocp-allowed-registries                                               FAIL     medium
upstream-ocp4-pci-dss-4-0-ocp-allowed-registries-for-import                                    FAIL     medium
% oc get ccr upstream-ocp4-pci-dss-4-0-container-security-operator-exists upstream-ocp4-pci-dss-4-0-security-profiles-operator-exists
NAME                                                           STATUS   SEVERITY
upstream-ocp4-pci-dss-4-0-container-security-operator-exists   PASS     medium
upstream-ocp4-pci-dss-4-0-security-profiles-operator-exists    PASS     medium
```

**Comment by rhmdnd** [comment]
> @yuumasato For rule security-operator-exists, there is no compliance.openshift.io/automated-remediation= label. And there is no complianceremediation generated for this rule when security profiles operator not installed.

If we look at the rule we have for FIO as an example, we don't provide an automatic remediation there either.

https://github.com/openshift/check-payload/pull/175/files#r1643499290

I think leaving this as an exercise for the user is good for now since they'll need to make some decisions about how they want to maintain it in their cluster (e.g., automatic updates versus manual updates).

**Review by rhmdnd — APPROVED** [review]
/lgtm

**Comment by yuumasato** [comment]
`Gate / Build, Test on CentOS 7 (Container) (pull_request)` finally passed.
Mergning per Dev and QE approvals.

**Pull request merged**